### ISSUE TITLE
Exploiting Crafting

### ISSUE THREAD

**Issue by Samikana** [DarkstarProject/darkstar#3066]
**_I have:_**

- [x] searched existing issues (http://github.com/darkstarproject/darkstar/issues/) to see if the issue I am posting has already been addressed or opened by another contributor
- [x] checked the commit log to see if my issue has been resolved since my server was last updated

**_Client Version_** (type `/ver` in game) **:** 30160329_1

**_Server Version_** (type `@revision` in game) **:** a8cf5b83cee234867b72a44fe4b9c1d789f86cfc

**_Source Branch_** (master/stable) **:** Stable (Told by TeoTwawki)

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** You can use addons and plugins to be notified the moment you start a craft if it's NQ/HQ/breakage/etc.  While crafting, you can use @ tele, @ homepoint. @ warp, or have someone cast Warp II or teleport you mid-craft if it's either going to break or NQ and you want HQ, etc.  This doesn't consume your materials and you potentially can just use the exact same mats.  The only thing it consumes is the crystal.

**Comment by Hozu** [comment]
I swore this was fixed before by making the synth break on being zoned. Though, what happens on retail is you lose the materials and crystal as soon as the synth starts. This is what causes people losing their Lu Shang's rod on retail, trying to fix it on the boat and the boat ride ends mid synth.

**Comment by TeoTwawki** [comment]
@Hozu I thought that too and started hunting through old commits looking for it. I also thought someone reported this same thing awhile back and it was determined this can't be happening. I can't find a commit or an older issue report now though. But I could swear I've seen it!

**Comment by teschnei** [comment]
you mean the ingredients are lost when you start, and the finished product is awarded at synth completion?

**Comment by Hozu** [comment]
Yeah, I'm pretty sure that's the case. It explains the loss of everything on zoning that retail has (unless they changed that sometime in the past 7 years or so).

**Comment by teschnei** [comment]
that sounds pretty easy to implement, though i think in your inventory screen they are just greyed out

**Comment by TeoTwawki** [comment]
yeah

**Comment by Deadwing888** [comment]
This fix may also fix the issue wherein items don't get lost on break, but instead stay greyed out in inventory.

**Comment by eraffxi** [comment]
i brought this up once before, i vaguely remember someone committing somethign about it but at the time i just added exceptions to the scripts for every spell i could think of to check the animation of the target before transporting

**Comment by maxtherabbit** [comment]
the code currently treats zoning during syntheses as a standard break, with the normal chance to lose materials

is that wrong?

**Comment by Hozu** [comment]
IIRC yes. I remember people being able to lose their broken lu shang's rod when they zone on the boat during the repair synth. All materials get lost.

**Comment by Hozu** [comment]
Unless it was changed in the post-75 era it would be all materials + crystal removed on synth start, then add the finished product on synth completion.
